OSATAnnotation.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  OSATAnnotation.swift
+//  OSAT-VideoCompositor
+//
+//  Created by hdutt on 17/12/22.
+//
+
+import AVFoundation
+
+/// OSATAnnotationProtocol  is a base protocol for all Annotation object layers supported by OSAT-VideoCompositor
+protocol OSATAnnotationProtocol {
+    func getAnnotationLayer() -> CALayer
+}
+
+/// OSATImageAnnotation holds data for  the image annotation to be rendered on Video
+public struct OSATImageAnnotation: OSATAnnotationProtocol {
+    let image: UIImage
+    let frame: CGRect
+    let timeRange: CMTimeRange
+    let caption: NSString?
```

Comment:
@HemDutt any reason we are using **NSString** and not just **String**?

Reply:
fixed